neurox/data/writer.py

```diff
@@ -151,11 +153,14 @@ def open(self, num_layers):
         if self.decompose_layers:
             for layer_idx in self.layers:
                 local_filename = f"{self.filename[:-5]}-layer{layer_idx}.{self.filename[-4:]}"
-                _writer = self.base_writer(local_filename)
+                if self.base_writer == HDF5ActivationsWriter:
+                    _writer = self.base_writer(local_filename, dtype=self.dtype)
+                else:
+                    _writer = self.base_writer(local_filename)
```

Comment:
Given that JSONWriter now takes dtype as well, we don't need this if anymore right?